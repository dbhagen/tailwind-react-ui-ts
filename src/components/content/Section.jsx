import React from 'react'
import PropTypes from 'prop-types'

import ContentBlock from './ContentBlock'

const Section = ({ is, ...rest }) => <ContentBlock is={is} {...rest} />

Section.propTypes = {
  is: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]),
}

Section.defaultProps = {
  is: 'section',
}

export default Section
