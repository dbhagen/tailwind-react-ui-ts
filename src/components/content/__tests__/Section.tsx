import React from 'react'
import { shallow } from 'enzyme'

import Section from '../Section'

const setup = (testProps = {}) => {
  const props = Object.assign(testProps)

  const wrapper = shallow(<Section {...props}>Lorem ipsum...</Section>)

  return {
    props,
    wrapper,
  }
}

describe('Section', () => {
  it('renders matching snapshot', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })
})
