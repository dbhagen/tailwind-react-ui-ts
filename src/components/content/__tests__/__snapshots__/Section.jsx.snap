// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders matching snapshot 1`] = `
<WithTheme(ContentBlock)
  is="section"
>
  Lorem ipsum...
</WithTheme(ContentBlock)>
`;
